### PR TITLE
applications: matter: Avoid provider duplicates on the recovery list

### PR DESCRIPTION
**Pull request by ArekBalysNordic**
We need to ensure that the BLE provider doesn't exist on the recovery list to avoid putting it there once again.

### CODE REVIEW

File: doc/nrf/releases_and_maturity/releases/release-notes-changelog.rst [hunk 1]
```diff
@@ -125,6 +125,11 @@ Applications
 
 This section provides detailed lists of changes by :ref:`application <applications>`.
 
+* Matter bridge application
+
+   * Added the :kconfig:option:`CONFIG_BRIDGE_BT_MAX_SCANNED_DEVICES` kconfig option to set the maximum amount of scanned Bluetooth LE devices.
```
Comment (wiba-nordic) [inline doc/nrf/releases_and_maturity/releases/release-notes-changelog.rst:130]:
```suggestion
   * Added the :kconfig:option:`CONFIG_BRIDGE_BT_MAX_SCANNED_DEVICES` kconfig option to set the maximum number of scanned Bluetooth LE devices.
```

File: doc/nrf/releases_and_maturity/releases/release-notes-changelog.rst [hunk 2]
```diff
@@ -125,6 +125,11 @@ Applications
 
 This section provides detailed lists of changes by :ref:`application <applications>`.
 
+* Matter bridge application
```
Comment (wiba-nordic) [inline doc/nrf/releases_and_maturity/releases/release-notes-changelog.rst:128]: Please move this to the Matter bridge section (around Line 170)

**Review by wiba-nordic — CHANGES_REQUESTED** [review]

File: samples/matter/common/src/bridge/ble_connectivity_manager.cpp [hunk 3]
```diff
@@ -883,8 +883,35 @@ BLEBridgedDeviceProvider *BLEConnectivityManager::Recovery::GetProvider(sys_slis
 	return provider;
 }
 
+bool BLEConnectivityManager::Recovery::EntryExists(BLEBridgedDeviceProvider *provider, sys_slist_t *list)
```
Comment (markaj-nordic) [inline samples/matter/common/src/bridge/ble_connectivity_manager.cpp:886]: Could be `static`, as well as `PutProvider` and `GetProvider`.

File: samples/matter/common/src/bridge/ble_connectivity_manager.cpp [hunk 4]
```diff
@@ -883,8 +883,35 @@ BLEBridgedDeviceProvider *BLEConnectivityManager::Recovery::GetProvider(sys_slis
...
+		SYS_SLIST_FOR_EACH_NODE_SAFE (list, node, tmpNodeSafe) {
+			ListItem* item = reinterpret_cast<ListItem *>(node);
+
+			if (!item) {
+				return false;
+			}
+
+			bt_addr_le_t addr = provider->GetBtAddress();
+			bt_addr_le_t storedAddr = item->mProvider->GetBtAddress();
+			if (memcmp(&addr, &storedAddr, sizeof(addr)) == 0) {
```
Comment (markaj-nordic) [inline samples/matter/common/src/bridge/ble_connectivity_manager.cpp:901]: We could use `bt_addr_le_cmp()` utility.

**Review by markaj-nordic — APPROVED** [review]

File: doc/nrf/releases_and_maturity/releases/release-notes-changelog.rst (hunk 2 above)
Comment (ArekBalysNordic) [reply doc/nrf/releases_and_maturity/releases/release-notes-changelog.rst:128]: Oh, I forgot that we have such a section :)

**Review by wiba-nordic — APPROVED** [review]

**Pull request merged**